telemetry/specs/telemetry_report.mdk

```diff
@@ -439,54 +456,78 @@ The Destination Port is user configurable. The expectation is that the same
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-      |RepType| InType| Report Length |   MD Length   |D|Q|F|I|  Rsvd |    
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
-      |            RepMdBits          |      Domain Specific ID       |  R
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  E
-      |            DSMdBits           |          DSMdstatus           |  P                        
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  O
-      |          Variable Optional Baseline & DS Metadata             |  R                  
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  T
-      |    Truncated Packet or  Additional DS Extension Data or TLV   |  | 
-      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
-      
+## Telemetry Reports Outer Header (Ver 2.0) (8 octets)
+
+The Telemetry Reports Outer Header immediately follows the UDP header
+whose destination port identifies the contents as a telemetry report.
+There is at most one instance of the Telemetry Reports Outer Header in a
+packet.
+
+```
+   0                   1                   2                   3
+   0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+  |  Ver  |   hw_id   |           Sequence Number                 |
+  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+  |                            Node ID                            |
+  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
+
 * Ver (4b): Version
-  : This specification defines **version 2.0**.
+  : This specification defines **version 2**.
 
 * hw_id (6b): Hardware ID
-  : Identifies the hardware subsystem within the network device that generated
+  : Identifies the hardware subsystem within the node that generated
     this report. For example, in a chassis with multiple linecards this could
     identify a specific linecard, or a subsystem within a linecard. The hw_id
-    is unique within the scope of a Node id.
+    is unique within the scope of a Node ID.
 
 * Sequence Number (22b): Sequence Number
   : Reflects the sequence of reports from a specific combination of
-    (Node id, hw_id) to a particular telemetry report destination. This
+    (Node ID, hw_id) to a particular telemetry report destination. This
     can be used to detect loss of telemetry reports before they reach their
     intended destination.
 
 * Node ID (32b): Node ID
-  : The unique ID of a network device (such as routers, switches, NICs etc.) This is generally administratively assigned. Node IDs
-    must be unique within a management domain.
+  : The unique ID of a node. This is generally administratively assigned.
+    Node IDs must be unique within a management domain.
+
+## Individual Report Header (Ver 2.0) (4+ octets) { #individual-report }
+
```

Comment:
Can we spell out the rationale behind this report coalescing? This is to reduce the # of report pkts in the network etc. IMO, "coalescing" is a keyword but it's mentioned only once in the change log at the end of this spec.

Reply:
Added a new subsection "Coalescing A Group of Telemetry Reports In A Single Packet", at the end of "Key Concepts".